repositories/ingested_data_read_repository.go

```diff
@@ -55,10 +71,10 @@ func createQueryDbForField(ctx context.Context, exec Executor, readParams models
 			readParams.Path[0], models.NotFoundError)
 	}
 
-	firstTableObjectId, err := getFirstTableObjectIdFromPayload(readParams.ClientObject, link.ChildFieldName)
+	firstTableLinkValue, err := getParentTableJoinField(readParams.ClientObject, link.ChildFieldName)
```

Comment:
the function and the logic within was wrong, partly because it was incorrectly named